tracing-appender/src/rolling.rs

```diff
@@ -558,17 +562,68 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            max_files,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, max_files: usize) {
+        let files = fs::read_dir(&self.log_directory).map(|dir| {
+            dir.filter_map(Result::ok)
+                .filter(|entry| {
+                    entry.file_name().to_string_lossy().contains(
+                        &self
+                            .log_filename_prefix
+                            .clone()
+                            .expect("prefix is always present"),
+                    )
+                })
+                .filter_map(|entry| {
+                    let created = entry
+                        .metadata()
+                        .and_then(|metadata| metadata.created())
+                        .ok()?;
+                    Some((entry, created))
+                })
+                .collect::<Vec<_>>()
+        });
+        let mut files = match files {
+            Ok(files) => files,
+            Err(error) => {
+                eprintln!("Error reading the log directory/files: {}", error);
+                return;
+            }
+        };
+        if files.len() < max_files {
+            return;
+        }
+
+        // sort the files by their creation timestamps.
+        files.sort_by_key(|(_, created_at)| created_at);
```

Comment:
ah, my bad, it looks like my suggested change makes the borrow checker unhappy: https://github.com/tokio-rs/tracing/actions/runs/3114778302/jobs/5050994522
fortunately,  `SystemTime` is `Copy`, so we can just write this:
```suggestion
        files.sort_by_key(|(_, &created_at)| created_at);
```

Reply:
yeah I already fixed that no worries :)